spec/lib/tasks/cleanup_spec.rb

```diff
@@ -6,25 +6,36 @@
   include_context "rake"
 
   describe "cleanup:transient_registrations" do
+    let(:old_registration) { create(:new_registration, created_at: 31.days.ago) }
+    let(:env_limit) { 'TRANSIENT_REGISTRATION_CLEANUP_LIMIT' }
+
     before do
-      @old_registration = create(:new_registration, created_at: 31.days.ago)
+      old_registration
       create(:new_registration, created_at: 31.days.ago)
       create(:new_registration)
     end
 
-    it "deletes only up to the limit number of old transient registrations" do
-      expect { subject.invoke("1") }.to change {
-        [WasteExemptionsEngine::TransientRegistration
+    it "deletes only up to the environment variable limit number of old transient registrations" do
+      ENV[env_limit] = '1'
```

Comment:
ENV is global so it's best not to assign values to it. Prefer something like:
`allow(ENV).to receive(:fetch).with(env_limit).and_return("1")`